COVIDWatch iOS/ViewControllers/Confirm.swift

```diff
@@ -0,0 +1,122 @@
+
+    public var testedDate: Date = Date()
+    var ogSlideButtonPosition: CGPoint = CGPoint.zero
+    var slideStartPosition: CGFloat = 0.0
+    var slideEndPosition: CGFloat = 0.0
+    var slideCurrentPosition: CGFloat = 0.0
+    var btnWidth: CGFloat = 0.0
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        slideButton.layer.cornerRadius = 4
+        slideButton.layer.borderWidth = 4
+        slideButton.layer.borderColor = UIColor.Primary.White.cgColor
+        slideButton.addGestureRecognizer(UIPanGestureRecognizer(target: self, action: #selector(onSlide)))
+        btnWidth = slideButton.bounds.width
+
+        slideView.layer.cornerRadius = 10
+        slideStartPosition = slideView.bounds.minX + 8
+        let sliderWidth = slideView.bounds.width - 16
+        slideEndPosition = slideStartPosition + sliderWidth
+
+        boxView.layer.cornerRadius = 10
+        boxView.layer.borderWidth = 1
+        boxView.layer.borderColor = UIColor.Secondary.LightGray.cgColor
+
+        cancelButton.layer.cornerRadius = 10
+        cancelButton.layer.borderWidth = 2
+        cancelButton.layer.borderColor = UIColor.Secondary.LightGray.cgColor
+    }
+
+    @objc func onSlide(gesture: UIPanGestureRecognizer) {
+        if let gView = gesture.view,
+            let xPos = gesture.view?.center.x,
+            let yPos = gesture.view?.center.y {
+
+            if gesture.state == .began {
+                ogSlideButtonPosition = gView.center
+            }
+            if gesture.state == .began || gesture.state == .changed {
+                let translation = gesture.translation(in: slideView)
+                let viewFrame = slideView.convert(gView.frame, from: slideButton)
+                let currentLeftPos = viewFrame.minX + translation.x
+                slideCurrentPosition = xPos + translation.x + (btnWidth / 2.0)
+                let slideDistance = (slideCurrentPosition - ogSlideButtonPosition.x)
+                let totalSlideDistance = screenWidth - btnWidth - 94
+                slideToConfirmText.alpha = 1.0 - 2 * (slideDistance / totalSlideDistance)
+                if currentLeftPos > slideStartPosition && slideDistance < totalSlideDistance {
+                    gesture.view?.center = CGPoint(x: xPos + translation.x, y: yPos)
+                    gesture.setTranslation(CGPoint.zero, in: slideView)
+                } else if slideDistance >= totalSlideDistance {
+                    onConfirm()
+                }
+            }
+            if gesture.state == .ended {
+                gesture.view?.center = ogSlideButtonPosition
+                gesture.setTranslation(CGPoint.zero, in: slideView)
+                slideToConfirmText.alpha = 1.0
+            }
+        }
+    }
+
+    func onConfirm() {
+        UserDefaults.shared.isUserSick = true
+        UserDefaults.shared.lastTestedDate = Date()
+        performSegue(withIdentifier: "confirmToHome", sender: self)
+    }
+}
+
+extension Confirm {
```

Comment:
Is there a reason for this to be an extension? Can't it just go in the class?

Reply:
This is just a coding style in swift I've seen in a lot of projects to split up the class to have extensions that serve different purposes for easier readability. In the sense that the dynamic styles that are earliest in the lifecycle are in a separate place that is MARK'ed so others can easily get to it and understand the code. It helps keep the code modular rather than have a single gigantic class where you gotta search for the needle in a haystack.

Reply:
It doesn't make a difference when compiled, its just for readability